core/src/test/java/org/frankframework/pipes/IfPipeTest.java

```diff
@@ -0,0 +1,303 @@
+package org.frankframework.pipes;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.stream.Stream;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.core.PipeForward;
+import org.frankframework.core.PipeRunException;
+import org.frankframework.core.PipeRunResult;
+import org.frankframework.stream.Message;
+import org.frankframework.util.CloseUtils;
+
+/**
+ * Contains generic {@link IfPipe} scenarios to test
+ */
+public class IfPipeTest extends PipeTestBase<IfPipe> {
```

Comment:
I'd like to see some tests where both jsonPath and xPath expressions are configured, and using the expected path-expression based on type of input.